crates/valence_generated/build/status_effects.rs

```diff
@@ -0,0 +1,227 @@
+use heck::ToPascalCase;
+use proc_macro2::TokenStream;
+use quote::quote;
+use serde::Deserialize;
+use valence_build_utils::{ident, rerun_if_changed};
+
+#[derive(Deserialize, Debug)]
+pub enum StatusEffectCategory {
+    Beneficial,
+    Harmful,
+    Neutral,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct StatusEffect {
+    id: u16,
+    name: String,
+    translation_key: String,
+    category: StatusEffectCategory,
+    color: u32,
+    instant: bool,
+}
+
+pub fn build() -> anyhow::Result<TokenStream> {
+    rerun_if_changed(["extracted/effects.json"]);
+
+    let effects =
+        serde_json::from_str::<Vec<StatusEffect>>(include_str!("../extracted/effects.json"))?;
+
+    let effect_count = effects.len();
+
+    let effect_from_raw_id_arms = effects
+        .iter()
+        .map(|effect| {
+            let id = &effect.id;
+            let name = ident(effect.name.to_pascal_case());
+
+            quote! {
+                #id => Some(Self::#name),
+            }
+        })
+        .collect::<TokenStream>();
+
+    let effect_to_raw_id_arms = effects
+        .iter()
+        .map(|effect| {
+            let id = &effect.id;
+            let name = ident(effect.name.to_pascal_case());
+
+            quote! {
+                Self::#name => #id,
+            }
+        })
+        .collect::<TokenStream>();
+
+    let effect_from_ident_arms = effects
+        .iter()
+        .map(|effect| {
+            // TODO: put the full resource identifier in the extracted JSON.
```

Comment:
Does this TODO need to be resolved in this PR?

Reply:
I copied the TODO from `sound.rs`

Reply:
I would say no

Reply:
Ok, then lets remove that comment.

Reply:
> Ok, then lets remove that comment.

alright